language/move-lang/src/shared/mod.rs

```diff
@@ -227,3 +278,44 @@ pub fn format_delim<T: fmt::Display, I: IntoIterator<Item = T>>(items: I, delim:
 pub fn format_comma<T: fmt::Display, I: IntoIterator<Item = T>>(items: I) -> String {
     format_delim(items, ", ")
 }
+
+//**************************************************************************************************
+// Flags
+//**************************************************************************************************
+
+pub type Flags = Vec<Flag>;
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum Flag {
+    Test,
+}
+
+impl Flag {
+    pub const KNOWN_FLAGS: &'static [Flag] = &[Flag::Test];
+
+    pub fn long_flag_name(&self) -> &str {
+        match self {
+            Flag::Test => "test",
+        }
+    }
+
+    pub fn short_flag_name(&self) -> &str {
+        match self {
+            Flag::Test => "t",
+        }
+    }
+
+    pub fn parse_str(flag_str: &str) -> Result<Flag, &str> {
+        match flag_str {
+            "t" | "test" => Ok(Flag::Test),
+            _ => Err(flag_str),
+        }
```

Comment:
Still feels a little duplicatedy with struct opt. My main concern here is what happens if you do `move-check --help` or anything like that. I would want these flags to be displayed. Additionally I would want this logic to be shared in with the move-cli. I think the best way of doing that might be to make another struct with a bag of options or something instead of this enum. Something like 
```
#[derive(StructOpt)]
struct Flags {
  #[structopt(global = true, default_value = false)]
  test: bool, 
}
```

Reply:
In short doing anything but struct_opt here feels wrong, as it will not integrate well with other end points that use move-lang. 

Even if you do something like `struct FlagOptions` and then translate that into `enum Flag` and `Set<Flag>`

Reply:
Ahh, I now realize what you were getting at yesterday. I'll update to this -- it makes a lot more sense than the current implementation.

Reply:
Should be updated now. Let me know what you think of the new flags definition.